src/workspaces.c

```diff
@@ -388,15 +388,79 @@ workspaces_find(struct workspace *anchor, const char *name, bool wrap)
 	return NULL;
 }
 
+static void
+destroy_workspace(struct workspace *workspace)
+{
+	wlr_scene_node_destroy(&workspace->tree->node);
+	zfree(workspace->name);
+	wl_list_remove(&workspace->link);
+	free(workspace);
+}
+
+void
+workspaces_reconfigure(struct server *server)
+{
+	/*
+	 * Currently, we consider updates like below as "WS2 is renamed to WS3
+	 * and WS2 is added":
+	 * old workspaces = ["WS1", "WS2"]
+	 * new workspaces = ["WS1", "WS3", "WS2"]
+	 */
+	struct workspace *conf;
+	struct wl_list *ws_link = server->workspaces.next;
```

Comment:
Consider `s/ws_link/elm/`

The actual `wl_list` macros/functions use `elm` in places. We've also done the same in a few instances.
https://github.com/nobled/wayland/blob/f5ccd7a9e670e097ff9b5faae8a26b3af4bb7b60/src/wayland-util.h#L112

Reply:
I don't think `elm` is a good name here.
`elm` seems fine if it is used only in `for(;;)` or `wl_container_of(...)`, but it is used throughout this function here and `elm` lacks its context.

I replaced it with `actual_workspace_link` instead.